### PR TITLE
Move from /deploy to /config

### PR DESCRIPTION
**Pull request by fmuyassarov**
Moves `/deploy/tls/`  & `/deploy/basic-auth` under `/config` folder. Also updates the documentation with references to the config folder instead of deploy folder.

### CODE REVIEW

File: config/ironic_ci.env [hunk 1]
```diff
@@ -0,0 +1,11 @@
+HTTP_PORT=6180
```
Comment (maelk) [inline config/ironic_ci.env:1]: maybe that file can go under hack folder instead ?

File: docs/deploying.md [hunk 2]
```diff
@@ -14,38 +14,68 @@ Operator(BMO) with or without Ironic as well as deploying only Ironic scenario.
...
+│   ├── kustomizeconfig.yaml
+│   └── patches
+│       ├── cainjection_in_baremetalhosts.yaml
+│       └── webhook_in_baremetalhosts.yaml
+├── default
+│   ├── kustomization.yaml
+│   ├── manager_auth_proxy_patch.yaml
+│   ├── manager_webhook_patch.yaml
+│   └── webhookcainjection_patch.yaml
 ├── ironic_ci.env
```
Comment (maelk) [inline docs/deploying.md:43]: remove ?

**Review by maelk — COMMENTED** [review]
looks good, just two small things

File: config/ironic_ci.env (hunk 1 above)
Comment (fmuyassarov) [reply config/ironic_ci.env]: Yep, done

File: docs/deploying.md (hunk 2 above)
Comment (fmuyassarov) [reply docs/deploying.md]: Done, thanks.

**Review by maelk — APPROVED** [review]

**Review by furkatgofurov7 — COMMENTED** [review]
Looks good to me!

**Pull request merged**